pallets/asset/src/lib.rs

```diff
@@ -450,6 +451,7 @@ decl_module! {
         #[weight = T::DbWeight::get().reads_writes(4, 3) + 500_000_000]
         pub fn register_ticker(origin, ticker: Ticker) -> DispatchResult {
             let sender = ensure_signed(origin)?;
+            CallPermissions::<T>::ensure_call_permissions(&sender)?;
```

Comment:
could be merged into one internal function and return the sender for success.

Reply:
E.g., `let sender = ensure_permissioned_signer(origin)?;`.